src/ray/raylet/local_object_manager.h

```diff
@@ -171,6 +204,29 @@ class LocalObjectManager {
   /// This class is accessed by both the raylet and plasma store threads. The
   /// mutex protects private members that relate to object spilling.
   mutable absl::Mutex mutex_;
+
+  ///
+  /// Fields below are used to delete spilled objects.
+  ///
+
+  /// A list of object id and url pairs that need to be deleted.
+  /// We don't instantly delete objects when it goes out of scope from external storages
+  /// because those objects could be still in progress of spilling.
+  std::queue<ObjectID> spilled_object_pending_delete_ GUARDED_BY(mutex_);
+
+  /// Mapping from object id to url_with_offsets. We cannot reuse pinned_objects_ because
+  /// pinned_objects_ entries are deleted when spilling happens.
+  absl::flat_hash_map<ObjectID, std::string> spilled_objects_url_ GUARDED_BY(mutex_);
+
+  /// Base URL -> ref_count. It is used because there could be multiple objects
+  /// within a single spilled file. We need to ref count to avoid deleting the file
+  /// before all objects within that file are out of scope.
+  absl::flat_hash_map<std::string, uint64_t> url_ref_count_ GUARDED_BY(mutex_);
+
+  /// The interval to process deletion queue.
+  uint32_t delete_interval_ms_ = 100;
```

Comment:
Can we reuse the existing `free_objects_period_ms_` and `free_objects_batch_size_`?

Reply:
Sounds good. Btw, this will delete objects every 1 second. Do you think it isn't too slow?